src/base/utils/gzip.cpp

```diff
@@ -31,127 +31,190 @@
 
 #include <vector>
 
+#include <QtGlobal>
+#include <QBuffer>
 #include <QByteArray>
+#include <QIODevice>
 
 #ifndef ZLIB_CONST
 #define ZLIB_CONST  // make z_stream.next_in const
 #endif
 #include <zlib.h>
 
-QByteArray Utils::Gzip::compress(const QByteArray &data, const int level, bool *ok)
+bool Utils::Gzip::compress(QIODevice &source, QIODevice &dest, int level)
```

Comment:
Another concern. It seems Windows 10 doesn't support opening .gz files with GUI and at least 3rd-party app must be installed. So I would suggest you compress it to .zip files instead.

Note that windows 10 do support .gz files via native command line `tar` but I won't expect novice users to use it.

Reply:
> So I would suggest you compress it to .zip files instead.

Another dependency?
In any case, I would not consider this idea as a requirement. In other words, `Windows 10 doesn't natively support opening .gz files with GUI and at least 3rd-party app is required` does not seem to me a sufficient reason to reject this feature in its current form.
Adding support for `zip` archives is a separate issue (while `gzip` is already supported).

Reply:
>In any case, I would not consider this idea as a requirement. In other words, Windows 10 doesn't natively support opening .gz files with GUI and at least 3rd-party app is required does not seem to me a sufficient reason to reject this feature in its current form.

IMO it is bad user experience. Some novice user enabled compress log option (GUI doesn't give any information about the format) and now cannot deal with it without googling or installing other tools.

Anyway I've already raised my concern. Also since this option is off by default, I don't mind to turn a blind eye about it. Yet I would still hope to see that the GUI will inform the user about the format.

Offtopic: in linux world, there exist this tool: [logrotate](https://linux.die.net/man/8/logrotate) and it is prevalent. It seems to me that this PR does share part of the functionality with it (assuming logrotate can work with qbt logs) which more or less made me think that the point is moot for linux users.

Reply:
> IMO it is bad user experience. Some novice user enabled compress log option (GUI doesn't give any information about the format) and now cannot deal with it without googling or installing other tools.

IMO, "bad user experience" is not the feature itself, but only "GUI doesn't give any information about the format". So...


> I would still hope to see that the GUI will inform the user about the format.

:+1:

Reply:
> in linux world, there exist this tool: [logrotate](https://linux.die.net/man/8/logrotate) and it is prevalent. It seems to me that this PR does share part of the functionality with it (assuming logrotate can work with qbt logs) which more or less made me think that the point is moot for linux users.

Well, this means that Linux users may (possibly) have an alternative to this built-in functionality of qBittorrent.

Reply:
After reading such a statement, someone might think that the feature is not supported (not available) on Windows without 3rd-party tools.

Reply:
> There should be no ambiguity now.

Tooltip looks good for me now. But I would change `Compress log files with gzip when backing up` with `Compress backup logs with gzip` to be consistent with other options about backups.

BTW, I think it's better to use `log backups` instead of `backup logs`.